### PR TITLE
✨ ci: collect debug information about leftover processes

### PR DESCRIPTION
**Pull request by chrischdi**
**What this PR does / why we need it**:

Collects information about leftover processes for ci jobs.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:

Helps on debugging #8560

### CODE REVIEW

File: scripts/ci-e2e.sh [hunk 1]
```diff
@@ -90,6 +90,14 @@ cleanup() {
...
   ps -ef > "${ARTIFACTS_LOCAL}/processes-ps-ef.txt" || true
 
+  for PID in $(ps -eo pid=); do
+    echo "> PID=$PID"
+    echo ">> /proc/${PID}/status" 
+    cat "/proc/${PID}/status" || true
+    echo ">> /proc/${PID}/stack" 
+    cat "/proc/${PID}/stack" || true
+  done >> "${ARTIFACTS_LOCAL}/processes-proc-information.txt"
+
```
Comment (sbueringer) [inline scripts/ci-e2e.sh:100]: Q: Will this fail the ci-e2e.sh script when we hit something like this?
```bash
cat: /proc/306131/status: No such file or directory
cat: /proc/306131/stack: No such file or directory
```

File: scripts/ci-e2e.sh [hunk 2]
```diff
@@ -90,6 +90,14 @@ cleanup() {
 
   ps -ef > "${ARTIFACTS_LOCAL}/processes-ps-ef.txt" || true
 
+  for PID in $(ps -eo pid=); do
+    echo "> PID=$PID"
+    echo ">> /proc/${PID}/status" 
+    cat "/proc/${PID}/status" || true
+    echo ">> /proc/${PID}/stack" 
+    cat "/proc/${PID}/stack" || true
+  done >> "${ARTIFACTS_LOCAL}/processes-proc-information.txt"
```
Comment (sbueringer) [inline scripts/ci-e2e.sh:99]: On my machine that is 1,4 MB of data. I assume it's significantly less in the Pod at this time, but let's keep an eye on it

Comment (sbueringer) [reply scripts/ci-e2e.sh:99]: Could have just checked in e2e-main... 12483 bytes seems okay

Comment (chrischdi) [reply scripts/ci-e2e.sh:99]: we could also compress the data. But yes, should be way less due to way less pids inside the prowjob pod.

**Pull request merged**